### PR TITLE
[Web-App] Target Reward: connected to the backend 

### PR DESCRIPTION
**Pull request by vitto-moz**
https://www.notion.so/saladtech/Add-Add-to-Chopping-List-Button-to-the-Reward-Page-with-request-d10b569b0cce403fa3c46f3a0f48d5fc?pvs=4
https://www.notion.so/saladtech/Get-the-initial-data-about-the-Target-Reward-and-update-it-according-to-the-update-rules-2-days-7ba83b81ff3c4f7985e10103dd1c8f3f?pvs=4
https://www.notion.so/saladtech/Connect-to-backend-65a726abb63b4ea0a16512b3e301599e?pvs=4
https://www.figma.com/file/neYpmmJ5Y0cKjltKeEE3Si/App-Library?node-id=5223-78336&t=vgaAI6nAsua8QSGa-4
image

### CODE REVIEW

**Review by seniorquico — CHANGES_REQUESTED** [review]

File: packages/web-app/src/modules/reward-views/components/RewardHeaderBar.tsx [hunk 1]
```diff
@@ -221,6 +229,9 @@ class _RewardHeaderBar extends Component<Props> {
                 </div>
               )}
             </div>
+            <Button className={classes.targetThisRewardButton} onClick={() => {}}>
```
[the remark this answers is not in the document]
Comment (vitto-moz) [reply packages/web-app/src/modules/reward-views/components/RewardHeaderBar.tsx:232]: there was not the functionality to handle the button click at the time the PR was created.
I have to check current progress to refine if we have for now..

File: packages/web-app/src/modules/home-views/NavigationBarContainer.tsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { Avatar, AvatarDefault, BonusCard, NavigationBar } from '@saladtechnologies/garden-components'
+import type { TargetRewardInfo } from '@saladtechnologies/garden-components/lib/components/NavigationBar/components/DesktopNavigationBar/TargetRewardStatus'
```
Comment (rners01) [inline packages/web-app/src/modules/home-views/NavigationBarContainer.tsx:2]: doesn't index reexport this to the top?
```suggestion
import type { TargetRewardInfo } from '@saladtechnologies/garden-components'
```

File: packages/web-app/src/modules/home-views/NavigationBarContainer.tsx [hunk 3]
```diff
@@ -16,6 +17,18 @@ const mapStoreToProps = (store: RootStore): any => {
...
+
+  const targetReward: TargetRewardInfo | undefined = selectedReward
+    ? {
+        id: selectedReward.id,
+        label: selectedReward.name,
+        imageSrc: selectedReward.coverImage ?? '',
+        price: selectedReward.price,
+        canBeRedeemed: store.balance.currentBalance >= selectedReward?.price,
+      }
+    : undefined
```
Comment (rners01) [inline packages/web-app/src/modules/home-views/NavigationBarContainer.tsx:30]: Its no big deal, but we defined that like so
image

File: packages/web-app/src/modules/home-views/NavigationBarContainer.tsx [hunk 4]
```diff
@@ -52,11 +65,13 @@ const mapStoreToProps = (store: RootStore): any => {
     startButtonClick: startButton.onClick,
     startButtonHoverLabel: undefined,
     startButtonErrorClick: startButton.onClickWithError,
+    onRemoveTargetRewardClick: store.rewards.removeSelectedReward,
     startButtonProgress: startButton.progress,
     startButtonRunningTime: startButton.runningTime,
     startButtonToolTip: startButton.toolTip,
     startButtonToolTipError: startButton.toolTipError,
     username: isAuthenticated ? store.profile.currentProfile?.username : undefined,
+    targetReward: targetReward,
```
Comment (rners01) [inline packages/web-app/src/modules/home-views/NavigationBarContainer.tsx:74]:
```suggestion
    targetReward,
```

File: packages/web-app/src/modules/reward-views/RewardDetailsContainer.tsx [hunk 5]
```diff
@@ -31,9 +32,9 @@ const mapStoreToProps = (store: RootStore, props: RouteComponentProps<{ id: stri
     reward: reward,
     onBack: store.routing.goBack,
     onRedeem,
-    isInCart: store.rewards.isInChoppingCart(props.match.params.id),
-    onAddToCart: store.rewards.addToChoppingCart,
-    onRemoveFromCart: store.rewards.removeFromChoppingCart,
```
Comment (rners01) [inline packages/web-app/src/modules/reward-views/RewardDetailsContainer.tsx:36]: why it was removed?

File: packages/web-app/src/modules/reward-views/RewardDetailsContainer.tsx [hunk 6]
```diff
@@ -22,6 +22,7 @@ const mapStoreToProps = (store: RootStore, props: RouteComponentProps<{ id: stri
     reward?.tags?.includes('requires-minecraft-username') && !extensions?.minecraftUsername
   const requiresPayPalAccount = reward?.tags?.includes('requires-paypal-account') && !hasPayPalAccount
   const requiresSaladCard = reward?.tags?.includes('requires-saladcard-account') && !hasSaladCard
+  const isTargetReward = store.rewards.selectedReward?.id === reward?.id
```
Comment (rners01) [inline packages/web-app/src/modules/reward-views/RewardDetailsContainer.tsx:25]:
```suggestion
  const isRewardSelectedAsTargetReward = store.rewards.selectedReward?.id === reward?.id
```

File: packages/web-app/src/modules/reward/RewardStore.tsx [hunk 7]
```diff
@@ -138,18 +138,23 @@ export class RewardStore {
     return this.rewards.get(id)
   }
 
-  isInChoppingCart = (id?: string): boolean => {
-    return this.selectedRewardId === id
+  @computed
+  public get selectedReward(): Reward | undefined {
```
Comment (rners01) [inline packages/web-app/src/modules/reward/RewardStore.tsx:142]:
```suggestion
  public get selectedTargetReward(): Reward | undefined {
```

File: packages/web-app/src/modules/reward/RewardStore.tsx [hunk 8]
```diff
@@ -138,18 +138,23 @@ export class RewardStore {
...
+    if (this.selectedRewardId) {
```
Comment (rners01) [inline packages/web-app/src/modules/reward/RewardStore.tsx:143]:
```suggestion
    if (this.selectedTargetRewardId) {
```

File: packages/web-app/src/modules/reward/RewardStore.tsx [hunk 9]
```diff
@@ -138,18 +138,23 @@ export class RewardStore {
...
   @action.bound
   loadSelectedReward = flow(function* (this: RewardStore) {
     var res = yield this.axios.get('/api/v1/profile/selected-reward')
+    yield this.loadReward(res.data.rewardId)
     this.selectedRewardId = res.data.rewardId
   })
 
   @action.bound
-  addToChoppingCart = flow(function* (this: RewardStore, reward: Reward) {
+  setSelectedReward = flow(function* (this: RewardStore, reward: Reward) {
```
Comment (rners01) [inline packages/web-app/src/modules/reward/RewardStore.tsx:157]: and in other cases
```suggestion
  setTargetSelectedReward = flow(function* (this: RewardStore, reward: Reward) {
```

**Comment by Maks19** [comment]
I found a bag with the first reward render
It shows Remove As Target Reward button firstly

https://user-images.githubusercontent.com/41080668/228028280-7485147b-c81c-4e8b-99d5-3c3aa2d676de.mp4

File: packages/web-app/src/modules/home-views/NavigationBarContainer.tsx (hunk 2 above)
Comment (vitto-moz) [reply packages/web-app/src/modules/home-views/NavigationBarContainer.tsx:2]: unfortunately the import you have suggested is not working

File: packages/web-app/src/modules/home-views/NavigationBarContainer.tsx (hunk 3 above)
Comment (vitto-moz) [reply packages/web-app/src/modules/home-views/NavigationBarContainer.tsx:30]: agree, wonder we don't have this spotted with TS compiler/linter

**Comment by vitto-moz** [comment]
> I found a bag with the first reward render It shows Remove As Target Reward button firstly

resolved

File: packages/web-app/src/modules/reward-views/components/RewardHeaderBar.tsx [hunk 10]
```diff
@@ -111,13 +119,13 @@ interface Props extends WithStyles<typeof styles> {
   authenticated?: boolean
   onBack?: () => void
   onRedeem?: (reward?: Reward) => void
-  isInCart?: boolean
-  onAddToCart?: (reward: Reward) => void
-  onRemoveFromCart?: (reward: Reward) => void
+  isTargerReward?: boolean
```
Comment (Maks19) [inline packages/web-app/src/modules/reward-views/components/RewardHeaderBar.tsx:122]:
```suggestion
  isTargetReward?: boolean
```

**Review by Maks19 — DISMISSED** [review]

**Review by Maks19 — APPROVED** [review]

**Pull request merged**